### PR TITLE
Restrict Google Analytics tag to not show link for restricted/under-review data

### PR DESCRIPTION
**Pull request by aimalkhan**
## What this PR accomplishes
Google Analytics tag shows resource's id regardless of whether the dataset is open or not. This change checks for user access before including the resource's id info into the page.

## Issue(s) addressed

* check user permissions and data access level before including resource id in the google analytics tag to the dataset page

## What needs review
* does making restricted/under-review pages not have resource tags mess with analytics, and should the tag be still included but after removing the resource information?
* check dataset page for this comment to see the relevant part of google analytics tag:
`// Pushed when a dataset page or a resource page is viewed`
* on a restricted/under-review access dataset page, the resource id is only included if the user is logged in. It does not appear if the user is not logged in
* on an open access dataset page, the resource id is included for all users

### CODE REVIEW

**Comment by KatiRG** [comment]
>     * does making restricted/under-review pages not have resource tags mess with analytics, and should the tag be still included but after removing the resource information?
> 
>     * check dataset page for this comment to see the relevant part of google analytics tag:
>       `// Pushed when a dataset page or a resource page is viewed`

You have set `this_pkg = ''` if the user does not have access to the resource page, which means that GA will not send any information about a visit to the dataset page because the length of `this_pkg` will not be > 0:

```
// Pushed when a dataset page or a resource page is viewed
    if ('{{this_pkg}}'.length > 0) {
        window.dataLayer.push({
            'event': 'Pageview',
            'ministryTag': '{{this_org}}',
            'group': '{{this_group}}',
            'dataAvailability': '{{this_pkg.access_level}}',
            'datasetName': '{{this_pkg.title}}',
            'resourceName': resourceName
        });
    } 
```
You should rather set `this_res` to `''` if the user has no access.

**Comment by KatiRG** [comment]
>     

- [ ] * on an open access dataset page, the resource id is included for all users

**Comment by aimalkhan** [comment]
`this_pkg` was the issue, as it has the resource id on the dataset page.
`this_res` is only populated on the resource page, and is empty already on the dataset, so it is not the issue.
to keep GA, `this_pkg` will need to be modified to remove the resource id; this what I will look into

File: ckanext/ontario_theme/templates/internal/gtm_pageview.html [hunk 1]
```diff
@@ -13,13 +13,14 @@
     }
 
     // Pushed when a dataset page or a resource page is viewed
-    if ('{{this_pkg}}'.length > 0) {
+    if ('{{this_pkg_access_level}}'.length > 0 ||
+        '{{this_pkg_title}}'.length > 0) {
```
Comment (KatiRG) [inline ckanext/ontario_theme/templates/internal/gtm_pageview.html:17]: why are you checking length of pkg title?

File: ckanext/ontario_theme/templates/internal/gtm_pageview.html [hunk 2]
```diff
@@ -13,13 +13,14 @@
     }
 
     // Pushed when a dataset page or a resource page is viewed
-    if ('{{this_pkg}}'.length > 0) {
+    if ('{{this_pkg_access_level}}'.length > 0 ||
```
[the remark this answers is not in the document]
Comment (aimalkhan) [reply ckanext/ontario_theme/templates/internal/gtm_pageview.html:16]: seems like it is a mandatory field 
image

File: ckanext/ontario_theme/templates/internal/gtm_pageview.html (hunk 1 above)
Comment (aimalkhan) [reply ckanext/ontario_theme/templates/internal/gtm_pageview.html:17]: previous check was `this_pkg`, which contained both `access_level` and `title`. This was an attempt to recreate the same check. I see the point that either being > 0 will mean that both are valid.

File: ckanext/ontario_theme/templates/internal/gtm_pageview.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (KatiRG) [reply ckanext/ontario_theme/templates/internal/gtm_pageview.html:16]: Actually, access level must be one of the 3 allowed options and can never be blank, therefore this check is not needed.

[the remark this answers is not in the document]
Comment (aimalkhan) [reply ckanext/ontario_theme/templates/internal/gtm_pageview.html]: fixed.

**Pull request merged**